docs/development/pigs/pig-009.rst

```diff
@@ -0,0 +1,238 @@
+    events = irf_sampler.sample_edisp(events)
+
+Alternatively the IRFs could be applied per model component and be handled in the `MapEventSampler`. However in this case
+the IRFs would not be applied correctly for e.g. very extended sources, because the IRF changes over region of the source.
+If we find the approach with the `IRFMapSampler` to be not performant enough or problematic in memory use, it can be still
+used as a simple fallback solution.
+
+MapDatasetEventSampler?
+=======================
+
+To combine the `MapEventSampler` and `IRFMapSampler` objects we finally propose to introduce a `MapDatasetEventSampler`,
+which executes the sampling from all source components, applies the IRFs, samples from the background model and returns
+a stacked, combined event list.
+
+This is the proposed high level API:
+
+.. code::
+    
+    dataset_sampler = MapDatasetEventSampler(dataset)
+    events = dataset_sampler.sample()
+
+This requires to add `t_min` and `t_max` information to `MapDataset` or maybe a `MapDataset.gti` attribute, defining the
+GTI for the given dataset. In addition it also requires to add a lightcurve model to the `SkyModel`.
+
+More detailed:
+
+.. code::
+
+    events = []
+
+    # sample from source components
+    for evaluator in dataset._evaluators:
+        evaluator.psf = None?
+        evaluator.edisp = None?
+        npred = evaluator.compute_npred()
+
+        sampler = MapEventSampler(npred)
+
+        # set an ID for the source component?
+        
+        events.append(sampler.sample())
+
+    events_stacked = vstack(events)
+
+    # apply IRFs
+    irf_sampler = IRFSampler(events_stacked, dataset.psf_map, dataset.edisp_map)
+    events_reco = irf_sampler.sample()
+
+    # sample from background
+    bkg_sampler = MapEventSampler(dataset.background_model.map)
+    events_bkg = bkg_sampler.sample()
+
+    events_total = vstack([events_reco, event_bkg])
+
+
+Simulating multiple observations
+================================
+The question of simulating mutiple observations from e.g. an `ObservationTable` or `GTI` object as it is needed for
+simulating data for the CTA data challenge is not addressed in this PIG.
+
+
+
+Alternatives
+============
+Instead of introducing a parallel class structure of sampler objects once could introduce `.sample()` methods on the
+existing related classes in Gammapy:
+
+- `events_src = MapEvaluator.sample()` to sample from a source model
+- `events = EdispMap.sample(events=)` to apply the energy dispersion
+- `events = PsfMap.sample(events=)` to apply the PSF
+- `events = MapDataset.sample()` to combine all together.
+
+This has the advantage of a smaller API, but is less flexible.
+
+
+Binned Simulation
+-----------------
+As an alternative approach one could also use binned simulations by fluctuating the number of counts in each bin of a
+given map or spectrum. This is not a real alternative, but could be offered in addition and used depending on the use case.
```

Comment:
I would suggest to put "binned simulation" in the introduction of the PIG (or as a sentence after introduction).
It's not really an alternative. We already have binned simulation, and it will remain:
- https://docs.gammapy.org/0.11/api/gammapy.spectrum.SpectrumSimulation.html
- https://docs.gammapy.org/0.11/notebooks/simulate_3d.html

If you prefer to keep the mention of binned simulation here at the end, OK, but please at least mention more clearly that preliminary implementations of binned simulation in Gammapy exist and that feature will remain, i.e. don't start the paragraph mentioning it with "An alternative approach .."